tensorboard/plugins/core/core_plugin.py

```diff
@@ -93,9 +96,25 @@ def _serve_asset(self, path, gzipped_asset_bytes, request):
     return http_util.Respond(
         request, gzipped_asset_bytes, mimetype, content_encoding='gzip')
 
+  @wrappers.Request.application
+  def _serve_data_location(self, request):
```

Comment:
Hmm, every endpoint we add is another request the frontend has to make to the backend on load.  Given that the data returned is so small, could we maybe instead start consolidating under some common endpoint like "/data/properties" or "/data/environment" or whatever, and fold "/data/window_properties" into that endpoint?

Then going forward we can include other small pieces of data that are known immediately at startup (versus dynamic data like runs, etc.) in that shared endpoint rather than having to support a whole bunch of different endpoints.  E.g. at some point including the version in there would be nice so we can display the version somewhere in the GUI (perhaps under the settings menu, or even just logged to the JS console, but some way that we can detect it from the browser would be useful in debugging).

Reply:
+1, added /data/environment. Indeed, that would save extra requests for base properties, and we'll probably add more in the future.